File: Creational/FactoryMethod/PHP/factories/WindowsWindow.php
<?php

require_once(__DIR__.'/../interfaces/IButton.php');
require_once(__DIR__.'/../products/WinButton.php');
require_once('AbstractWindow.php');

/**
 * Concrette factory.
 */

class WindowsWindow extends AbstractWindow
{
	/**
	 * @override
	 */
	protected function createButton(): IButton
	{
		return new WinButton();
	}
}

File: Creational/FactoryMethod/Java/factories/WindowsWindow.java
package patterns.creational.factorymethod.factories;

import patterns.creational.factorymethod.interfaces.IButton;
import patterns.creational.factorymethod.products.WinButton;

/**
 * Concrete factory.
 */

public class WindowsWindow extends AbstractWindow
{
	/**
	 * @Override
	 */
	protected IButton createButton()
	{
		return new WinButton();
	}
}
